controllers/charmedk8scontrolplane_controller.go

```diff
@@ -237,10 +286,43 @@ func (r *CharmedK8sControlPlaneReconciler) Reconcile(ctx context.Context, req ct
 // SetupWithManager sets up the controller with the Manager.
 func (r *CharmedK8sControlPlaneReconciler) SetupWithManager(mgr ctrl.Manager) error {
 	return ctrl.NewControllerManagedBy(mgr).
+		WithOptions(controller.Options{
+			RateLimiter: workqueue.NewMaxOfRateLimiter(
```

Comment:
This uses the same rate limiting setup provided with the default limiter, but uses a lower max value for the exponential limiter, 5 minutes instead of 16 minutes